docs/design/proposals/labels.md

```diff
@@ -0,0 +1,109 @@
+    - Currently we dont differentiate, it results in a conflict (https://github.com/kro-run/kro/pull/631)
+    - To indicate that an instance is being actively reconciled against a specific RGD. This label will be applied to the instance itself.
+    - **Proposed Labels:**
+      - `kro.run/managed-by-rgd: my-rgd-name` - name of the ResourceGraphDefinition that reconciles the instance.
+    - **Existing Labels:**
+      - `kro.run/resource-graph-definition-id: 761e8fb7-14d5-4da1-b4f9-0a16fc334d6c`
+      - `kro.run/resource-graph-definition-name: my-rgd-name`
+      - Conflicts with label used in creation path for nested RGD (https://github.com/kro-run/kro/pull/631)
+4. **Kubernetes Common Labels:**
+    - **Purpose:** Conforming to standard/common k8s labels allows easier integration across tools in k8s ecosystem.
+    - **Proposed Labels:**
+      - `app.kubernetes.io/managed-by: KRO` : Fixed value
+
+## Other solutions considered
+
+1. We could continue using the existing labels, but their purpose is not as explicit for relationship tracking, which can lead to confusion for users and client tools.
+2. We could also use annotations, but labels are better suited for this purpose as they are queryable via the Kubernetes API, which is a key requirement for observability and tooling.
+3. Use common k8s labels: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/
+  * `app.kubernetes.io/name:	mysql`
+  * `app.kubernetes.io/instance: mysql-abcxyz`
+  * `app.kubernetes.io/version: 5.7.21`
+  * `app.kubernetes.io/component: database`
+  * `app.kubernetes.io/part-of: wordpress`
+  * `app.kubernetes.io/managed-by: Helm`
+
+
+## Scoping
+
+#### What is in scope for this proposal?
+
+-  Defining and adding the new labels to the KRO API.
+-  Updating the instance controller to apply these labels to instances and created resources during reconciliation.
+-  Updating the official KRO documentation to reflect the new labels and their usage.
+-  Adding docs reflecting the new labels. Old labels would not be documented.
+-  Deprecate old labels in next minor release or the one after that.
+
+#### What is not in scope?
+
+- Tracking UID od RGD, Instance in the labels. We need to handle recreation of parent resources where UID changes. Can be a future enhancement.
+- Supporting deeply nested RGDs. When RGD1 created RGD2 and that creates RGD3, do we add labels in objects of RGD3 linking it back to RGD2 and RGD1 ?
+- Changing how KRO uses annotations.
+- Changing these existing labels:
+  - kro.run/kro-version: v0.4.0
+  - kro.run/owned: "true"
+
+## Testing strategy
+
+#### Requirements
+
+-   A running Kubernetes cluster to deploy KRO and test the label application.
+
+#### Test plan
+
+-   **Unit tests:** Add unit tests for any new labeler functions in `pkg/metadata/labels.go` to ensure they construct the correct labels.
+-   **Integration tests:**
```

Comment:
we may want a migration test, to ensure we don't mess up deployed objects with the new version.

We may also want to ensure that in case users jumps directly from the current labels layout to the proposed one (i.e. skip the transition versions) kro keeps running as expected